includes/definitions/discovery/audiocodes.yaml

```diff
@@ -0,0 +1,153 @@
+mib: AC-ANALOG-MIB:AcGateway:AC-PSTN-MIB:DS1-MIB:AC-PM-PSTN-MIB
+modules:
+    sensors:
+        pre-cache:
+            data:
+                - oid:
+                    - callerDisplayCallerDisplay
+                    - channelsStartingPhoneNum
+                    - acAnalogFxsFxoType
+                    - acTrunkDialPlanName
+                    - dsx1CircuitIdentifier
+                    - acPMTrunkUtilizationAttributes
+        state:
+            data:
+                -
+                    oid: acAnalogFxsFxoTable
+                    value: acAnalogFxsFxoHookState
+                    num_oid: .1.3.6.1.4.1.5003.9.10.1.4.1.1.18.{{ $index }}
+                    descr: 'FXO {{ $index }} ({{ $callerDisplayCallerDisplay }}, {{ $channelsStartingPhoneNum }})'
+                    index: 'hookStatus{{ $index }}'
+                    state_name: acAnalogFxsFxoHookState
+                    states:
+                        - { descr: offHook, graph: 0, value: 0, generic: 0 }
+                        - { descr: onHook, graph: 0, value: 1, generic: 0 }
+                    skip_values:
+                        -
+                            oid: acAnalogFxsFxoType
+                            op: '!='
+                            value: 0
+                -
+                    oid: acAnalogFxsFxoTable
+                    value: acAnalogFxsFxoHookState
+                    num_oid: .1.3.6.1.4.1.5003.9.10.1.4.1.1.18.{{ $index }}
+                    descr: 'FXS {{ $index }} ({{ $callerDisplayCallerDisplay }}, {{ $channelsStartingPhoneNum }})'
+                    index: 'hookStatus{{ $index }}'
+                    state_name: acAnalogFxsFxoHookState
+                    states:
+                        - { descr: offHook, graph: 0, value: 0, generic: 0 }
+                        - { descr: onHook, graph: 0, value: 1, generic: 0 }
+                    skip_values:
+                        -
+                            oid: acAnalogFxsFxoType
+                            op: '!='
```

Comment:
If off-hook the sensor will be deleted during discovery.

Reply:
I filter on acAnalogFxsFxoType, not on acAnalogFxsFxoHookState. So the discovery is not impacted at all iwth HookState. 
This bloc is 3 times the same sensor declare (based on acAnalogFxsFxoType) but depending on the device, I create it with FXO, FXS or E&M description.